kornia/geometry/line.py

```diff
@@ -0,0 +1,173 @@
+# kornia.geometry.line module inspired by Eigen::geometry::ParametrizedLine
+# https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/ParametrizedLine.h
+from typing import Optional, Union
+
+import torch
+
+from kornia.core import Module, Parameter, Tensor, normalize
+from kornia.geometry.linalg import squared_norm
+from kornia.testing import KORNIA_CHECK, KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+
+__all__ = ["ParametrizedLine", "fit_line"]
+
+
+# TODO: implement me: https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h
+class Hyperplane:
+    """Not implemented yet: https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h."""
+
+    pass
+
+
+class ParametrizedLine(Module):
+    """Class that describes a parametrize line.
+
+    A parametrized line is defined by an origin point :math:`o` and a unit
+    direction vector :math:`d` such that the line corresponds to the set
+
+    .. math::
+
+        l(t) = o + t * d
+    """
+
+    def __init__(self, origin: Tensor, direction: Tensor) -> None:
+        """Initializes a parametrized line of direction and origin.
+
+        Args:
+            origin: the origin point of the line of any dimension.
+            direction: the normalized vector direction of any dimension.
+
+        Example:
+            >>> o = torch.tensor([0.0, 0.0])
+            >>> d = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine(o, d)
+        """
+        super().__init__()
+        self._origin = Parameter(origin)
+        self._direction = Parameter(direction)
+
+    def __str__(self) -> str:
+        return f"Origin: {self.origin}\nDirection: {self.direction}"
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    @property
+    def origin(self) -> Tensor:
+        """Return the line origin point."""
+        return self._origin
+
+    @property
+    def direction(self) -> Tensor:
+        """Return the line direction vector."""
+        return self._direction
+
+    def dim(self) -> int:
+        """Return the dimension in the line holds."""
+        return len(self.direction)
+
+    @classmethod
+    def through(cls, p0, p1) -> "ParametrizedLine":
+        """Constructs a parametrized line going from a point :math:`p0` to :math:`p1`.
+
+        Args:
+            p0: tensor with first point :math:`(B, D)`.
```

Comment:
i do not understand the meaining of B and D

Reply:
D is the point number of dimensions

Reply:
and B is batch? Does parametrizedLines support batches?

Reply:
nope, we do. As soon as we don't implement any iterative method and we can vector is alright, which i think is the case for now